src/arbitrum/compounders/gmx/GlpCompounder.sol

```diff
@@ -75,15 +78,22 @@ contract GlpCompounder is TokenCompounderBase {
     /********************************** Mutated Functions **********************************/
 
     /// @notice Extending the base function to enable deposit of any one of GLP's underlying assets.
-    function depositUnderlying(address _underlyingAsset, uint256 _underlyingAssets, address _receiver, uint256 _minAmount) public nonReentrant returns (uint256 _shares) {
-        if (!(_underlyingAssets > 0)) revert ZeroAmount();
+    function depositUnderlying(address _underlyingAsset, uint256 _underlyingAmount, address _receiver, uint256 _minAmount) public payable nonReentrant returns (uint256 _shares) {
+        if (!(_underlyingAmount > 0)) revert ZeroAmount();
 
-        IERC20(_underlyingAsset).safeTransferFrom(msg.sender, address(this), _underlyingAssets);
+        if (_underlyingAsset == ETH) {
+            if (msg.value != _underlyingAmount) revert InvalidAmount();
+
+            _underlyingAsset = WETH;
+            IWETH(WETH).deposit{value: _underlyingAmount}();
```

Comment:
Where can I see how deposit works and how it handles the accounting of ETH/WETH? Is it just that ETH is removed from the message sender's balance and in turn WETH is being credited 1:1 which will then be used to transfer into the contract below?

Reply:
See WETH deposit function at L38 https://etherscan.io/token/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2#code - it simply credits the account with an ERC20 token equivalent to the `msg.value`.

Regarding seeing it in action, ill improve coverage of test suite as I wrote before, it will be there.